ext/crypto/00_crypto.js

```diff
@@ -149,6 +151,112 @@
     return StringPrototypeReplace(base64String, /=/g, "");
   }
 
+  function isArrayOrArrayBufferView(obj) {
+    return ArrayIsArray(obj) || ArrayBufferIsView(obj);
+  }
+
+  function concatBytes(...bytes) {
+    return TypedArrayFrom(
+      Uint8Array,
+      bytes.reduce(
+        (a, val) =>
+          (isArrayOrArrayBufferView(val)) ? [...a, ...val] : [...a, val],
+        [],
+      ),
+    );
+  }
+
+  function buildTLV(tag, value) {
+    if (isArrayOrArrayBufferView(value)) {
+      value = concatBytes(...value);
+    }
+    const len = value.length;
+    const lenBytes = [];
+
+    if (len <= 127) {
+      lenBytes.push(len);
+    } else if (len < 256) {
+      lenBytes.push(0x81);
+      lenBytes.push(len);
+    } else if (len < 65536) {
+      lenBytes.push(0x82);
+      lenBytes.push(len >> 8);
+      lenBytes.push(len & 0xff);
+    }
+
+    return concatBytes(tag, lenBytes, value);
+  }
+
+  function buildIntegerTLV(val) {
+    return buildTLV(0x02, (val[0] & 0x80) ? [0x00, ...val] : val);
+  }
+
+  function buildRsaPkcs1(key, type) {
+    const modExp = concatBytes(buildIntegerTLV(key.n), buildIntegerTLV(key.e));
+
+    if (type == "private") {
+      let keyData = concatBytes(
+        buildIntegerTLV([0x00]),
+        modExp,
+        buildIntegerTLV(key.d),
+      );
+
+      if (key.p) {
+        keyData = concatBytes(
+          keyData,
+          buildIntegerTLV(key.p),
+          buildIntegerTLV(key.q),
+          buildIntegerTLV(key.dp),
+          buildIntegerTLV(key.dq),
+          buildIntegerTLV(key.qi),
+        );
+      }
+
+      return buildTLV(0x30, keyData);
+    } else {
+      return buildTLV(0x30, modExp);
+    }
+  }
+
+  function convertJwkRsaKey(jwk) {
```

Comment:
As pointed out in https://github.com/denoland/deno/pull/12022#discussion_r707008417 The [`rsa::pkcs1::ToRsaPublicKey`](https://docs.rs/pkcs1/0.2.4/pkcs1/trait.ToRsaPublicKey.html) and it's corresponding private key trait offers exactly what you need. Example for public key:
```rust
pub struct JWK { ... };

impl ToRsaPublicKey for JWK {
  fn to_pkcs1_der(&self) -> Result<pkcs1::RsaPublicKeyDocument> {
    let pub_key = pkcs1::RsaPublicKey { 
      modulus: decode(self.n),
      public_exponent: decode(self.e),
    };

    Ok(pub_key.to_der())
  }
}
```